lib/wibox/layout/align.lua

```diff
@@ -41,121 +43,297 @@ local align = {}
-        size_second = self._private.dir == "y" and h or w
-        -- if all the space is taken, skip the rest, and draw just the middle
-        -- widget
-        if size_second >= size_remains then
-            return { base.place_widget_at(self._private.second, 0, 0, width, height) }
-        else
-            -- the middle widget is sized first, the outside widgets are given
-            --  the remaining space if available we will draw later
-            size_remains = floor((size_remains - size_second) / 2)
+    local is_vert = self._private.dir == "y"
+
+    -- Get the required size of a child widget
+    local function fit_child(child, max_size)
+        -- Can't fit a widget if it does not exist
+        if not self._private[child] then
+            return 0
         end
+
+        max_size = max_size and max_size or size_remains
+        local width_remains = is_vert and width or max_size
+        local height_remains = is_vert and max_size or height
+
+        local w, h = base.fit_widget(
+            self, context, self._private[child],
+            width_remains, height_remains
+        )
+        return is_vert and h or w
     end
-    if self._private.first then
-        local w, h, _ = width, height, nil
-        -- we use the fit function for the "inside" and "none" modes, but
-        --  ignore it for the "outside" mode, which will force it to expand
-        --  into the remaining space
-        if self._private.expand ~= "outside" then
-            if self._private.dir == "y" then
-                _, h = base.fit_widget(self, context, self._private.first, width, size_remains)
-                size_first = h
-                -- for "inside", the third widget will get a chance to use the
-                --  remaining space, then the middle widget. For "none" we give
-                --  the third widget the remaining space if there was no second
-                --  widget to take up any space (as the first if block is skipped
-                --  if this is the case)
-                if self._private.expand == "inside" or not self._private.second then
-                    size_remains = size_remains - h
-                end
-            else
-                w, _ = base.fit_widget(self, context, self._private.first, size_remains, height)
-                size_first = w
-                if self._private.expand == "inside" or not self._private.second then
-                    size_remains = size_remains - w
-                end
-            end
+    -- Place a child widget inside the layout
+    local function place_child(child, offset, size, update_size_remains)
+        -- Can't place a widget if it does not exist or there is no more size
+        if not self._private[child] or size_remains <= 0 then
+            return
+        end
+
+        offset = floor(offset)
+        if not size then
+            size = size and floor(size) or size_remains
```

Comment:
`size` is falsy in this branch, so the first test line 75 is always falsy. I'm not sure what `floor(size)` (aka `floor(falsy)`) would return here...

Reply:
It is a left over from when I was trying to make it without the `if` block, I didn't notice that should be changed. Updated it in [135369c](https://github.com/awesomeWM/awesome/pull/3588/commits/135369c8e472c084806079c2353349c27169aa85).